saml/config.go

```diff
@@ -60,55 +158,97 @@ func NewConfig(entityID, acs, issuer, metadata *url.URL) (*Config, error) {
 	return cfg, nil
 }
 
-// GenerateAuthRequestID generates an auth XSD:ID conform ID.
-// A UUID prefixed with an underscore.
-func GenerateAuthRequestID() (string, error) {
-	newID, err := uuid.GenerateUUID()
-	if err != nil {
-		return "", err
-	}
-
-	// Request IDs have to be xsd:ID, which means they need to start with an underscore or letter,
-	// which is not always given for UUIDs.
-	return fmt.Sprintf("_%s", newID), nil
-}
-
 // Validate validates the provided configuration.
 func (c *Config) Validate() error {
 	const op = "saml.Config.Validate"
 
-	if c.AssertionConsumerServiceURL == nil {
-		return fmt.Errorf("%s: ACS URL not set: %w", op, oidc.ErrInvalidParameter)
+	if c.AssertionConsumerServiceURL == "" {
+		return fmt.Errorf("%s: ACS URL not set: %w", op, ErrInvalidParameter)
 	}
-
-	if c.EntityID == nil {
-		return fmt.Errorf("%s: EntityID not set: %w", op, oidc.ErrInvalidParameter)
+	if _, err := url.Parse(c.AssertionConsumerServiceURL); err != nil {
+		return fmt.Errorf("%s: failed to parse ACS URL: %w", op, ErrInvalidParameter)
 	}
 
-	if c.Issuer == nil {
-		return fmt.Errorf("%s: Issuer not set: %w", op, oidc.ErrInvalidParameter)
+	if c.EntityID == "" {
+		return fmt.Errorf("%s: EntityID not set: %w", op, ErrInvalidParameter)
+	}
+	if _, err := url.Parse(c.EntityID); err != nil {
+		return fmt.Errorf("%s: failed to parse EntityID: %w", op, ErrInvalidParameter)
 	}
 
-	if c.MetadataURL == nil {
-		return fmt.Errorf("%s: Metadata URL not set: %w", op, oidc.ErrInvalidParameter)
+	if c.MetadataURL == "" && c.MetadataXML == "" && c.MetadataParameters == nil {
+		return fmt.Errorf("%s: One of MetadataURL, MetadataXML, or MetadataParameters "+
+			"must be set: %w", op, ErrInvalidParameter)
+	}
+	if c.MetadataURL != "" {
+		if _, err := url.Parse(c.MetadataURL); err != nil {
+			return fmt.Errorf("%s: failed to parse MetadataURL: %w", op, ErrInvalidParameter)
+		}
+	}
+	if c.MetadataXML != "" {
+		if _, err := parseIDPMetadata([]byte(c.MetadataXML)); err != nil {
+			return fmt.Errorf("%s: %s: %w", op, err.Error(), ErrInvalidParameter)
+		}
 	}
 
-	if c.ValidUntil == nil {
-		return fmt.Errorf("%s: ValidUntil func not provided: %w", op, oidc.ErrInvalidParameter)
+	if c.MetadataParameters != nil {
+		if err := c.MetadataParameters.Validate(); err != nil {
+			return fmt.Errorf("%s: %s: %w", op, err.Error(), ErrInvalidParameter)
+		}
 	}
 
 	if c.GenerateAuthRequestID == nil {
 		return fmt.Errorf(
-			"%s: GenerateAuthRequestID func not provided: %w",
+			"%s: DefaultGenerateAuthRequestID func not provided: %w",
```

Comment:
[non-blocking] Should this be just `GenerateAuthRequestID`?

Reply:
Whoops, done in https://github.com/hashicorp/cap/pull/84/commits/8b5e66cad3a49bf34478a922d1a1cf5945ab178e 👍